app/controllers/vendor_api/vendor_api_controller.rb

```diff
@@ -15,10 +18,30 @@ def parameter_missing(e)
       render json: { errors: [{ error: 'ParameterMissing', message: e }] }, status: 422
     end
 
-  private
-
     def set_cors_headers
       headers['Access-Control-Allow-Origin'] = '*'
     end
+
+    def require_valid_api_token!
+      return if valid_api_token?
+
+      unauthorized_response = {
+        errors: [
+          {
+            error: 'Unauthorized',
+            message: 'Please provide a valid authentication token',
+          },
+        ],
+      }
+
+      render json: unauthorized_response, status: 401
+    end
+
+    def valid_api_token?
+      return unless request.headers['Authorization']
+
+      unhashed_token = request.headers['Authorization'].split(' ').last
```

Comment:
Shall we insist on `Bearer`?

Reply:
Fixed by using _Rails_! https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/298/commits/36c71e86d37e4e4fbffee969ee839edc4645edf3

https://github.com/rails/rails/blob/66cabeda2c46c582d19738e1318be8d59584cc5b/actionpack/lib/action_controller/metal/http_authentication.rb#L331